### PR TITLE
Remove service_workers_support from api/FileReaderSync

### PR DESCRIPTION
**Pull request by hamishwillee**
There is a [Worker support guideline](https://github.com/mdn/browser-compat-data/blob/master/docs/data-guidelines.md#web-workers-worker_support) so `service_workers_support` is incorrect. This follows on from this discussion:  https://github.com/mdn/browser-compat-data/issues/7849#issuecomment-745445095

@ddbeck The way I have done this  is to just add a note indicating that service workers are not supported (with the version, if known). 

Are there other alternatives? i.e. should/can I do multiple worker supports like this?:
```
 "worker_support": {
        "__compat": {
          "description": "Available in workers",
...

 "worker_support": {
        "__compat": {
          "description": "Available in service workers",
...
````

Or something like this (feels confusing).
```json
 "worker_support": {
        "__compat": {
          "description": "Available in workers",
          "support": {
            "chrome": [
              {
                "version_added": true,
              },
              {
                "version_added": true,
                "version_removed": "59",
                "notes": "Service workers not supported from Chrome 59"
              },
            ],
```

### CODE REVIEW

**Comment by hamishwillee** [comment]
Note, same question for caches. In chrome, for example, support was added for service workers in version 40. In version 43 support was added for other workers. 

So I what is the right way to do this - added in version 40, and a note stating that initially only for service workers. Or two worker_support lines, or one worker support line but with notes about what happened in each version?

**Comment by ddbeck** [comment]
It's hard to follow the diff for removals and data changes in one commit. Can you put into words the support story you're trying to tell here? The diff and the examples given seem to be telling competing stories. What was supported (and unsupported) and in what versions did support change?

That said, I can rule out one thing: multiple `"worker_support"` features. JSON keys have to be unique, or else we'll just clobber one bit of data with another.

Data like this also seems unlikely:

```
            "chrome": [
              {
                "version_added": true,
              },
              {
                "version_added": true,
                "version_removed": "59",
                "notes": "Service workers not supported from Chrome 59"
              },
            ],
```

The second support statement says support was dropped in version 59. The note is a bit strange, since it reiterates what `"version_removed": "59"` is already saying: from version 59, the feature is unsupported (though the first support statement says that support was reintroduced in some unknown version).

**Comment by hamishwillee** [comment]
> Can you put into words the support story you're trying to tell here? 

Hi @ddbeck 
All I have done (see diff) is deleted the `service_workers_support` key and captured the version information about service worker support  in a note under `worker_support`.

So to answer your question, this is supposed to say:
- workers, other than service workers are supported
- Info of versions in which there is support for workers
- Info about versions in which there _was_ support for service workers.

This is probably OK for this case but I'm not sure it is the "right way". How would YOU present for chrome  this imaginary case.
- dedicated workers are supported from version 4, 
- shared workers are supported from version 8
- service workers are supported from version 2 to 6. 
- chrome workers are not supported

File: api/FileReaderSync.json [hunk 1]
```diff
@@ -249,103 +249,59 @@
           }
         }
       },
-      "service_workers_support": {
+      "worker_support": {
         "__compat": {
-          "description": "Service workers support",
+          "description": "Available in workers (except for service workers)",
```
Comment (ddbeck) [inline api/FileReaderSync.json:254]: The linter won't accept any description but "Available in workers" here (though I appreciate what you're getting at).

```suggestion
          "description": "Available in workers",
```

File: api/FileReaderSync.json [hunk 2]
```diff
@@ -249,103 +249,59 @@
...
           "support": {
             "chrome": {
               "version_added": true,
-              "version_removed": "59"
+              "notes": "Not supported in service workers from Chrome 59"
```
Comment (ddbeck) [inline api/FileReaderSync.json:258]: nit: style

```suggestion
              "notes": "From Chrome 59, not supported in service workers."
```

File: api/FileReaderSync.json [hunk 3]
```diff
@@ -249,103 +249,59 @@
...
             },
             "chrome_android": {
               "version_added": true,
-              "version_removed": "59"
+              "notes": "Not supported in service workers from Chrome 59"
```
Comment (ddbeck) [inline api/FileReaderSync.json:262]: nit: style

```suggestion
              "notes": "From Chrome 59, not supported in service workers."
```

File: api/FileReaderSync.json [hunk 4]
```diff
@@ -249,103 +249,59 @@
...
             "edge": {
-              "version_added": null
+              "version_added": "12",
+              "notes": "Not supported in service workers"
             },
             "firefox": {
-              "version_added": true,
-              "version_removed": "61"
+              "version_added": "8",
+              "notes": "Not supported in service workers from FF61"
```
Comment (ddbeck) [inline api/FileReaderSync.json:270]: nit: style

```suggestion
              "notes": "From Firefox 61, not supported in service workers."
```

File: api/FileReaderSync.json [hunk 5]
```diff
@@ -249,103 +249,59 @@
...
             },
             "chrome_android": {
               "version_added": true,
-              "version_removed": "59"
+              "notes": "Not supported in service workers from Chrome 59"
             },
             "edge": {
-              "version_added": null
+              "version_added": "12",
+              "notes": "Not supported in service workers"
```
Comment (ddbeck) [inline api/FileReaderSync.json:266]: nit

```suggestion
              "notes": "Not supported in service workers."
```

File: api/FileReaderSync.json [hunk 6]
```diff
@@ -249,103 +249,59 @@
...
-              "version_added": true,
-              "version_removed": "61"
+              "version_added": "8",
+              "notes": "Not supported in service workers from FF61"
             },
             "firefox_android": {
-              "version_added": true,
-              "version_removed": "61"
+              "version_added": "8",
+              "notes": "Not supported in service workers from FF61"
```
Comment (ddbeck) [inline api/FileReaderSync.json:274]: nit: style

```suggestion
              "notes": "Form Firefox 61, not supported in service workers."
```

File: api/FileReaderSync.json [hunk 7]
```diff
@@ -249,103 +249,59 @@
...
             },
             "ie": {
-              "version_added": false
+              "version_added": true,
+              "notes": "Not supported in service workers"
```
Comment (ddbeck) [inline api/FileReaderSync.json:278]: Maybe we should omit this note? If I remember correctly, IE doesn't support service workers in any case (e.g., doesn't implement `navigator.serviceWorker`)

```suggestion
              "version_added": true
```

File: api/FileReaderSync.json [hunk 8]
```diff
@@ -249,103 +249,59 @@
...
             },
             "opera": {
               "version_added": true,
-              "version_removed": "46"
+              "notes": "Not supported in service workers from Opera 46"
```
Comment (ddbeck) [inline api/FileReaderSync.json:282]: nit: style

```suggestion
              "notes": "From Opera 46, not supported in service workers."
```

File: api/FileReaderSync.json [hunk 9]
```diff
@@ -249,103 +249,59 @@
...
             },
             "opera_android": {
               "version_added": true,
-              "version_removed": "43"
+              "notes": "Not supported in service workers from Opera 43"
```
Comment (ddbeck) [inline api/FileReaderSync.json:286]: nit: style

```suggestion
              "notes": "From Opera 43, not supported in service workers."
```

File: api/FileReaderSync.json [hunk 10]
```diff
@@ -249,103 +249,59 @@
...
-              "version_added": true
-            },
-            "opera_android": {
-              "version_added": true
-            },
-            "safari": {
-              "version_added": true,
-              "version_removed": "6.1"
+              "version_removed": "6.1",
+              "notes": "Service workers not supported"
```
Comment (ddbeck) [inline api/FileReaderSync.json:291]: suggestion: for consistency with other notes in this PR

```suggestion
              "notes": "Not supported in service workers."
```

File: api/FileReaderSync.json [hunk 11]
```diff
@@ -249,103 +249,59 @@
...
             },
             "safari_ios": {
               "version_added": true,
-              "version_removed": "7"
+              "version_removed": "7",
+              "notes": "Service workers not supported"
```
Comment (ddbeck) [inline api/FileReaderSync.json:296]: suggestion: for consistency with other notes in this PR

```suggestion
              "notes": "Not supported in service workers."
```

File: api/FileReaderSync.json [hunk 12]
```diff
@@ -249,103 +249,59 @@
...
             },
             "samsunginternet_android": {
-              "version_added": true
+              "version_added": true,
+              "notes": "Not supported in service workers from version 7.0"
```
Comment (ddbeck) [inline api/FileReaderSync.json:300]:
```suggestion
              "notes": "From version 7.0, not supported in service workers."
```

File: api/FileReaderSync.json [hunk 13]
```diff
@@ -249,103 +249,59 @@
...
             },
             "webview_android": {
-              "version_added": true
+              "version_added": true,
+              "notes": "Not supported in service workers from version 59"
```
Comment (ddbeck) [inline api/FileReaderSync.json:304]: nit: style

```suggestion
              "notes": "From version 59, not supported in service workers."
```

**Review by ddbeck — CHANGES_REQUESTED** [review]
Thanks for the update, @hamishwillee.

First, to answer your hypothetical:

> How would YOU present for chrome this imaginary case.
> 
> * dedicated workers are supported from version 4,
> * shared workers are supported from version 8
> * service workers are supported from version 2 to 6.
> * chrome workers are not supported

Right now, given our schema, probably with something like this:

```json
{
  "version_added": "2",
  "notes": [
    "From version 8, shared workers are supported.",
    "From version 4, dedicated workers are supported.",
    "From version 2 to 6, service workers are supported; from version 7, service workers are unsupported."
  ]
}
```

Hypothetical aside, the new diff here makes a lot more sense to me. For example, in Chrome supported initially and generally, then dropped from service workers specifically. Suggests in line comments are just some style and consistency nits.

Also, just so I'm doing my due diligence here: I take it you did some sort of testing to figure out the support data here?

**Comment by hamishwillee** [comment]
> Hypothetical aside, the new diff here makes a lot more sense to me. For example, in Chrome supported initially and generally, then dropped from service workers specifically. Suggests in line comments are just some style and consistency nits.

Agree - what you have suggested makes sense for the current data.

NOTE, that support for workers implies support for dedicated workers, shared workers and service workers. It does not imply support for the {FF-only) chrome workers - which I propose we completely ignore.

> Also, just so I'm doing my due diligence here: I take it you did some sort of testing to figure out the support data here?

No. There is no new data in this change - it is just a reformatting of the data that was already present.

**Review by ddbeck — APPROVED** [review]
Looks good. Thank you! 🎉
